### PR TITLE
feat: blake2b integration

### PR DESCRIPTION
**Pull request by kevjue**

### CODE REVIEW

File: plonky2x/core/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ ci = []
 
 [dependencies]
 plonky2 = { git = "https://github.com/mir-protocol/plonky2.git", default-features = false }
-curta = { git = "https://github.com/succinctlabs/curta.git" }
+curta = { git = "https://github.com/succinctlabs/curta.git",  branch="kevjue/blake2b_refactor" }
```
Comment (ratankaliani) [inline plonky2x/core/Cargo.toml:17]: nit: change to main

File: plonky2x/core/src/frontend/hash/blake2/builder.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+use super::accelerator::BLAKE2BAccelerator;
+use super::digest_hint::BLAKE2BDigestHint;
+use super::proof_hint::BLAKE2BProofHint;
+use super::request::BLAKE2BRequest;
+use super::stark::{get_blake2b_data, stark};
+use crate::frontend::hint::synchronous::Async;
+use crate::prelude::*;
+
+impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
+    /// The constraints for an accelerated SHA computation using Curta.
```
Comment (ratankaliani) [inline plonky2x/core/src/frontend/hash/blake2/builder.rs:10]: nit: accelerated BLAKE

File: plonky2x/core/src/frontend/hash/blake2/curta.rs [hunk 3]
```diff
@@ -1,145 +1,94 @@
...
 
+use super::accelerator::BLAKE2BAccelerator;
+use super::request::BLAKE2BRequest;
+use super::stark::{compute_blake2b_last_chunk_index, digest_to_array};
 use crate::backend::circuit::PlonkParameters;
 use crate::frontend::vars::Bytes32Variable;
-use crate::prelude::{ByteVariable, CircuitBuilder, CircuitVariable, Variable};
+use crate::prelude::*;
 
 pub const MAX_NUM_CURTA_CHUNKS: usize = 1600;
```
Comment (ratankaliani) [inline plonky2x/core/src/frontend/hash/blake2/curta.rs:14]: Is there a reason we still need this?

File: plonky2x/core/src/frontend/hash/blake2/data.rs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        self.write_slice(&input.t_values);
+        self.write_slice(&input.end_bits);
+        self.write_slice(&input.digest_bits);
+        self.write_slice(&input.digest_indices);
+        self.write_slice(&input.digests);
+    }
+}
+
+impl<L: PlonkParameters<D>, const D: usize> ValueStream<L, D> {
+    /// Read sha input data from the stream.
```
Comment (ratankaliani) [inline plonky2x/core/src/frontend/hash/blake2/data.rs:68]: nit: blake

File: plonky2x/core/src/frontend/hash/blake2/stark.rs [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+            .iter()
+            .zip_eq(input.padded_chunks.chunks_exact(16))
+        {
+            println!(
+                "writing chunk: {:?}",
+                chunk_value
+                    .iter()
+                    .map(|x: &u64| u64_to_le_field_bytes::<L::Field>(*x))
+                    .collect_vec()
+            );
```
Comment (ratankaliani) [inline plonky2x/core/src/frontend/hash/blake2/stark.rs:68]: nit: rm?

File: plonky2x/core/src/frontend/hash/blake2/stark.rs [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+        }
+
+        writer.write(
+            &self.num_messages,
+            &L::Field::from_canonical_u32(input.digests.len() as u32),
+            0,
+        );
+    }
+
+    /// Generate a proof for the SHA stark given the input data.
```
Comment (ratankaliani) [inline plonky2x/core/src/frontend/hash/blake2/stark.rs:103]: nit: blake

**Review by ratankaliani — APPROVED** [review]
lgtm!

File: plonky2x/core/src/frontend/hash/blake2/curta.rs (hunk 3 above)
Comment (kevjue) [reply plonky2x/core/src/frontend/hash/blake2/curta.rs:14]: Good catch.  No, it's not needed.

**Pull request merged**